instrumentation/github.com/labstack/echo/otelecho/config.go

```diff
@@ -39,30 +38,29 @@ func (o optionFunc) apply(c *config) {
 	o(c)
 }
 
-// WithPropagators specifies propagators to use for extracting
-// information from the HTTP requests. If none are specified, global
-// ones will be used.
-func WithPropagators(propagators propagation.TextMapPropagator) Option {
-	return optionFunc(func(cfg *config) {
-		if propagators != nil {
-			cfg.Propagators = propagators
-		}
-	})
+// newConfig creates a new config struct and applies opts to it.
+func newConfig(opts ...Option) *config {
+	c := &config{}
+	for _, opt := range opts {
+		opt.apply(c)
+	}
+
+	return c
 }
 
-// WithTracerProvider specifies a tracer provider to use for creating a tracer.
-// If none is specified, the global provider is used.
-func WithTracerProvider(provider oteltrace.TracerProvider) Option {
-	return optionFunc(func(cfg *config) {
-		if provider != nil {
-			cfg.TracerProvider = provider
-		}
+// WithSkipper specifies a skipper for allowing requests to skip generating spans.
+func WithSkipper(skipper func(c echo.Context) bool) Option {
+	return optionFunc(func(conf *config) {
+		conf.otelhttpOptions = append(conf.otelhttpOptions, otelhttp.WithFilter(func(request *http.Request) bool {
```

Comment:
How does this interact if the user also provides an `otelhttp.WithFilter(...)` option?